### DISCUSSION TITLE
Are docket numbers unique?

### DISCUSSION THREAD

**Discussion by mlissner** [freelawproject/courtlistener#1623] [Q&A]
I'm trying to identify content in CourtListener, can I use docket numbers or something else to do so?

**Comment by mlissner** [comment]
Unfortunately, docket numbers are not unique in almost any sense. There are three major problems with them that come to mind and probably others that folks can chime in with:

1. They're used across courts.

    Many courts have similar approaches to docket numbers. For example, they might want to do the year-dash-some-number: 17-23829 or have a number like in a district court: 3:17-cv-00001-MJL (this would be civil case 1 from 2017 in office 3 by judge Michael Jay Lissner). 

2. Docket numbers can change
   
    Take the example above. The core docket number is 17-cv-00001, but if you're using the judge and office parts those can change if the judge is reassigned. Oops.

3. They have varied punctuation.

    This is workable, but computer folks run afoul of it. Because docket numbers are written by humans the punctuation is a mess. Horizontal lines in docket numbers can be either dashes, hyphens, emdashes, or endashes. If you're comparing docket numbers and somebody got an emdash in the middle of one but not the other, well, that won't match.

Also, remember that a single case can have multiple opinions or orders over time, so if you're looking in our opinion search engine, you might get multiple results for a single case.

-----

So you might be wondering how you identify a case uniquely. We've had that question many times over the years too and we usually take a heuristic approach to it. You can start with the docket number and jurisdiction. That's usually pretty good. If it's not enough, you can add the case name. Between the three you usually can get close, but you'll want to normalize docket numbers and case names. This is off topic, but case names can be written in many ways too, with abbreviations used or omitted, longer case names preferred or shorter ones, etc.

**Reply by troglodite2** [reply]
Case names change over time depending on when and where it is being heard.  Cases that are filed against Name in their official capacity as Governor of State might be known as Doe v. Gov1 when it starts and then it will be known as Doe v. Gov2 when Gov2 takes over.

In addition as cases move through the appeals process the names can be changed for other reasons.  Finally, cases sometimes revert back to their original name during other parts of the appeals process.
